src/compiler/checker.ts

```diff
@@ -11055,8 +11055,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 reportErrorsFromWidening(declaration, type);
             }
 
-            // always widen a 'unique symbol' type if the type was created for a different declaration.
-            if (type.flags & TypeFlags.UniqueESSymbol && (isBindingElement(declaration) || !declaration.type) && type.symbol !== getSymbolOfDeclaration(declaration)) {
+            // always widen a 'unique symbol' type if the type was created for a different declaration and if it isn't accessible
+            if (type.flags & TypeFlags.UniqueESSymbol && !declaration.type && type.symbol !== getSymbolOfDeclaration(declaration) && !isValueSymbolAccessible(type.symbol, type.symbol.valueDeclaration)) {

```

Comment:
I have a feeling that passing `type.symbol.valueDeclaration` to `isValueSymbolAccessible` here might not be 100% correct but so far I wasn't able to write a test that would fail with this